libkbfs/cr_chains.go

```diff
@@ -375,14 +386,22 @@ func (ccs *crChains) makeChainForOp(op op) error {
 			}
 		}
 
-		co := newCreateOp(realOp.NewName, ndu, realOp.RenamedType)
-		co.setWriterInfo(realOp.getWriterInfo())
-		co.renamed = true
-		co.Dir.Ref = ndr
-		err = ccs.addOp(ndr, co)
+		co, err := newCreateOp(realOp.NewName, ndu, realOp.RenamedType)
 		if err != nil {
 			return err
 		}
+		co.setWriterInfo(realOp.getWriterInfo())
+		co.renamed = true
+
+		// ndr may be zero if this is a post-resolution chain,
+		// so set co.Dir.Ref manually.
+		if ndr != (BlockPointer{}) {
+			co.Dir.Ref = ndr
+			err = ccs.addOp(ndr, co)
```

Comment:
Is there a reason to avoid calling `addOp` for the `createOp` case, but not for the `rmOp` case above?  Unless you know a bug with the old way of always calling `addOp`, it might be safer to keep it the way it was.